File: leetcode/0179-largest-number/solution.py
class Solution:
    def largestNumber(self, nums: list[int]) -> str:
        def compare(a, b) :
            if a+b > b+a :
                return -1
            elif b+a < a+b :
                return 1
            else :
                return 0

        answer = [str(x) for x in nums]
        answer.sort(key = cmp_to_key(compare))
        answer_str = ''.join(answer)
        
        return answer_str if answer_str[0] != "0" else "0" 

File: leetcode/0179-largest-number/Solution.java
class Solution {
    public String largestNumber(int[] nums) {
        String[] strs = new String[nums.length];
        String answer = "";
        for(int i = 0; i < nums.length; i++)
            strs[i] = Integer.toString(nums[i]);

        Arrays.sort(strs, (a, b) -> ((b+a).compareTo(a+b)));
        for (String str : strs) answer += str;
        
        if (answer.startsWith("0")) return "0";
        return answer;
    }
}